java-checks/src/main/java/org/sonar/java/checks/NestedSwitchStatementCheck.java

```diff
@@ -24,33 +24,29 @@
 import org.sonar.check.Rule;
 import org.sonar.plugins.java.api.IssuableSubscriptionVisitor;
 import org.sonar.plugins.java.api.tree.BaseTreeVisitor;
-import org.sonar.plugins.java.api.tree.CaseGroupTree;
-import org.sonar.plugins.java.api.tree.ClassTree;
 import org.sonar.plugins.java.api.tree.LambdaExpressionTree;
+import org.sonar.plugins.java.api.tree.NewClassTree;
 import org.sonar.plugins.java.api.tree.SwitchStatementTree;
 import org.sonar.plugins.java.api.tree.Tree;
-import org.sonar.plugins.java.api.tree.Tree.Kind;
 
 @Rule(key = "S1821")
 public class NestedSwitchStatementCheck extends IssuableSubscriptionVisitor {
 
   @Override
-  public List<Kind> nodesToVisit() {
+  public List<Tree.Kind> nodesToVisit() {
     return ImmutableList.of(Tree.Kind.SWITCH_STATEMENT);
   }
 
   @Override
   public void visitNode(Tree tree) {
-    SwitchStatementTree switchStatementTree = (SwitchStatementTree) tree;
-    List<CaseGroupTree> cases = switchStatementTree.cases();
-    cases.stream().forEach(c -> c.accept(new NestedSwitchVisitor()));
+    ((SwitchStatementTree) tree).cases().stream().forEach(c -> c.accept(new NestedSwitchVisitor()));
   }
 
   private class NestedSwitchVisitor extends BaseTreeVisitor {
 
     @Override
-    public void visitClass(ClassTree tree) {
-      // skip Class
+    public void visitNewClass(NewClassTree tree) {
```

Comment:
Sorry, I asked you to change the comment and **not the tree** you visit. Unfortunately, my comment was not complete... It should have been `skip nested classes and anonymous classes`.

You should revert that and continue skipping classes (ClassTree), otherwise you are going to have False Positives (FPs). The following (ugly) code is **now** raising FPs, while the previous one was not:
```java
  void gul(int i) {
    switch (i) {
      case 0:
      case 42:
        class B { // nested class
          void bar(int j) {
            switch (j) { // FP - this switch is not nested
              case 42:
                System.out.println("yolo");
                break;
              case 0:
                break;
              default:
                break;
            }
          }
        }
        new B().bar(i);
        break;

      default:
        break;
    }
  }
```